### ISSUE TITLE
Is it possible to disown automatically?

### ISSUE THREAD

**Issue by melboiko** [fish-shell/fish-shell#6021]
It is boring to type `disown` every time I want to exit a shell. (There are no situations where I want to kill background processes in the job list.  I want the processes to keep running in 100% of cases.)

Is there any way I can configure fish to run a `disown` automatically before exit (i.e. neither warn me that there are jobs in the background, nor require a second Ctrl+D, nor sent SIGHUPs?)

Thank you for your work!

version: 3.0.2
SO: Debian GNU/Linux 10 (buster)

**Comment by zanchey** [comment]
One method is to override the exit command:

```
function exit
    jobs -q; and disown (jobs -p)
    builtin exit
end
```

This will work if you type `exit`, or hit Ctrl-D, but not if you just close the terminal.

**Comment by zanchey** [comment]
So, I've been thinking about this some more, and I think there's definitely an argument for making this possible. No other shells do it though.

**Comment by mqudsi** [comment]
We could call `exit` when we get `SIGHUP` (when the terminal is closed), that would allow an override like the one you posted to work.

**Comment by mqudsi** [comment]
Giving this some more thought, that would only work if `exit` absolutely does not generate any input/output or else fish'll likely `SIGTTIN` or `SIGTTOU` and it'd just make everything worse.

There are still quite a few redirection bugs in fish, but the idea is that instead of calling `exit` on `SIGHUP` we can call `exit <&- >&- 2>&-` (or `exit <&- >/dev/null 2>/dev/null`), which should be safer.

**Comment by ridiculousfish** [comment]
Yeah I like that idea! `exit` should be executed on SIGHUP with closed inputs and outputs (which will prevent any jobs spawned by `exit` from just stopping)

**Comment by blastrock** [comment]
> No other shells do it though.

I think bash did it by default, and zsh does it with an option:

>  If you often start jobs that should go on even when the shell has exited, then you can set the option NO_HUP, and background jobs will be left alone. 

From the [zsh doc](https://zsh.sourceforge.io/Guide/zshguide02.html).

I regularly used this feature on zsh. However, as opposed to OP, I do like the double confirmation when exiting a shell with children. An option to enable this behavior would be nice.

**Comment by ramezquitao** [comment]
As @blastrock , I also use to close my terminals in bash and zsh and I like the background processes to keep running. I'm trying fish, and this is going to create a lot of problems for me until I get used to the situation. I also think that  an option to enable this behavior would be nice.

**Comment by ramezquitao** [comment]
Just tested in Debian 12 using fish version 3.6.0. 

Creating a function:

```
function on_exit --on-event fish_exit
    echo fish is now exiting
end
```
did the trick for ctrl-d and for closing the terminal (did the test running emacs &)

**Comment by teras** [comment]
> One method is to override the exit command:

Is this method still working?
For me, the process still is getting killed if I type ctrl-d or no exit is performed if I manually type `exit`.
